src/pykeen/nn/utils.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+"""Utilities for neural network components."""
+
+from typing import Optional, Sequence, Union
+
+import torch
+from more_itertools import chunked
+from torch import nn
+from tqdm.auto import tqdm
+
+__all__ = [
+    "TransformerEncoder",
+]
+
+
+class TransformerEncoder(nn.Module):
+    """A combination of a tokenizer and a model."""
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str,
+        max_length: Optional[int] = None,
+    ):
+        """
+        Initialize the encoder.
+
+        :param pretrained_model_name_or_path:
+            the name of the pretrained model, or a path, cf. :func:`transformers.AutoModel.from_pretrained`
+        :param max_length: >0, default: 512
+            the maximum number of tokens to pad/trim the labels to
+
+        :raise ImportError:
+            if the transformers library could not be imported
+        """
+        super().__init__()
+        try:
+            from transformers import AutoModel, AutoTokenizer
+        except ImportError as error:
+            raise ImportError("Please install the `transformers` library") from error
```

Comment:
@cthoyt we may add an optional requirement to the `setup.cfg`, and/or improve this error message.

Reply:
Done in 40cb229d